swift/apple/FirezoneKit/Sources/FirezoneKit/Views/UpdateNotification.swift

```diff
@@ -0,0 +1,245 @@
+  let apple: SemanticVersion
+}
+
+private class UpdateNotifier: NSObject, UNUserNotificationCenterDelegate {
+  @Published private(set) var decision: UNAuthorizationStatus
+  private var lastNotifiedVersion: SemanticVersion?
+  static let lastDismissedVersionKey = "lastDismissedVersion"
+  static let notificationIdentifier = "UPDATE_CATEGORY"
+  static let appStoreLink = URL(string: "https://apps.apple.com/us/app/firezone/id6443661826")!
+
+  override public init() {
+    self.decision = .notDetermined
+    super.init()
+
+    let notificationCenter = UNUserNotificationCenter.current()
+
+    // Define a custom action
+    let dismissAction = UNNotificationAction(identifier: "DISMISS_ACTION",
+                                             title: "Dismiss This Version",
+                                             options: [])
+
+    // Define a notification category with the action
+    let notificationCategory = UNNotificationCategory(identifier: UpdateNotifier.notificationIdentifier,
+                                                       actions: [dismissAction],
+                                                       intentIdentifiers: [],
+                                                       options: [])
+
+    notificationCenter.setNotificationCategories([notificationCategory])
+
+    notificationCenter.delegate = self
+    notificationCenter.requestAuthorization(options: [.sound, .badge, .alert]) { granted, error in
+      notificationCenter.getNotificationSettings { notificationSettings in
+        self.decision = notificationSettings.authorizationStatus
+      }
+    }
+
+  }
+
+  public func updateNotification(version: SemanticVersion) {
+    if let lastDismissedVersion = getLastDismissedVersion(), lastDismissedVersion >= version {
+      return
+    }
+
+    let content = UNMutableNotificationContent()
+    lastNotifiedVersion = version
+    content.title = "Update Firezone"
+    content.body = "New Firezone version available"
+    content.sound = .default
+    content.categoryIdentifier = UpdateNotifier.notificationIdentifier
+
+    let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
+
+    let uuidString = UUID().uuidString
+    let request = UNNotificationRequest(
+      identifier: uuidString, content: content, trigger: trigger
+    )
+
+    UNUserNotificationCenter.current().add(request) { error in
+      if let error = error {
+        Log.app.error("\(#function): Error requesting notification: \(error)")
+      }
+    }
+
+  }
+
+  func userNotificationCenter(_ center: UNUserNotificationCenter,
+                              didReceive response: UNNotificationResponse,
+                              withCompletionHandler completionHandler: @escaping () -> Void) {
+      if response.actionIdentifier == "DISMISS_ACTION" {
+        try? setLastDissmissedVersion(version: lastNotifiedVersion!)
+        return
+      }
+
+      NSWorkspace.shared.open(UpdateNotifier.appStoreLink)
```

Comment:
This link on the other hand could be inlined?

Reply:
it's also used in the menu bar, I think it's good to keep the same link.